[#10, #11] 유저를 조회한다.

gch01410 opened the pull request:
#10  해당 page 에 해당하는 number 개의 유저 list
#11 특정 유저 조회

조회 기능 한번에 pr 보냅니다.

Commit:
[#10] feat: 회원을 조회한다.

Commit:
[#10] feat: UserService에 회원조회 메서드를 추가한다.

Commit:
Merge remote-tracking branch 'upstream/develop' into feature/10/findUsers

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

Commit:
[#10] feat: UserInfoDto 를 구현한다.

- UserInfoDto : 유저 정보를 위한 Dto
- UserInfoDtoMapper : User 객체를 UserInfoDto 로 map 하는 객체

Commit:
[#10] refactor: 응답에서 UserInfoDto 를 사용하도록 한다.

Commit:
[#11] feat: 회원 id로 회원을 조회한다.

Commit:
[#11] feat: userService에 id로 회원을 조회하는 메소드를 구현한다.

TheDevLuffy commented on src/main/java/com/woowacourse/tecobrary/user/command/application/UserService.java:
```diff
@@ -52,4 +60,17 @@ private User findById(String userNo) {
+    }
+
+    public UserInfoDto findUserById(long id) {
+        User user = userRepository.findById(id).orElseThrow(NotFoundUserException::new);
```
```suggestion
        User user = userRepository.findById(id)
                .orElseThrow(NotFoundUserException::new);
```

gch01410 replied:
수정했습니다!

TheDevLuffy commented on src/main/java/com/woowacourse/tecobrary/user/command/util/UserInfoDtoMapper.java:
```diff
@@ -0,0 +1,28 @@
+
+public class UserInfoDtoMapper {
+
+    public static UserInfoDto map(User user) {
```
```suggestion
    public static UserInfoDto toDto(User user) {
```

TheDevLuffy commented on src/main/java/com/woowacourse/tecobrary/user/ui/UserController.java:
```diff
@@ -24,4 +36,17 @@ public ResponseEntity countOfUser(Model model) {
+    public ResponseEntity findUsers(@RequestParam int page, @RequestParam int number, Model model) {
+        List<UserInfoDto> users = userService.findUsersOnPage(page, number);
+        model.addAttribute("users", users);
+        return ResponseEntity.ok().body(model);
```
이슈 등록 필요
1. `findUsers` 메서드
    - model 을 사용하지 않고 ``ok(users)`` 로 변경합니다.
    - users 없이 inline 으로 리팩토링 합니다. => LibraryBookController 참고 !

2. ``findUserById`` 메서드
    - 위와 동일합니다.

TheDevLuffy commented on src/main/java/com/woowacourse/tecobrary/user/ui/dto/UserInfoDto.java:
```diff
@@ -0,0 +1,36 @@
+@NoArgsConstructor
+@Getter
+@ToString
+@EqualsAndHashCode
```
값을 꺼내는 방법으로 Test 를 수정하는 것이 어떨까요~?

gch01410 replied:
👍

TheDevLuffy commented on src/test/java/com/woowacourse/tecobrary/user/ui/UserControllerTest.java:
```diff
@@ -24,4 +26,80 @@ void successfullyCountOfUser() {
+
+    @DisplayName("[GET] /users?page=string&number=string, 적절하지 않은 쿼리 파라미터 값을 보내면 Bad Request 응답을 받는다.")
+    @Test
+    void failFindUsers2() {
```
```suggestion
    void failFindUsersInvalidParams() {
```

workromancer reviewed (changes requested):
안녕하세요 리뷰어 루피입니다.

코멘트 몇개 남겼으니 확인 부탁드려요~

gch01410 commented:
파라미터에 final 키워드 붙이기

Commit:
[#11] refactor: 컨벤션을 적용한다.

- UserInfoDtoMapper의 map 메서드명을 toDto로 변경한다.
- UserController의 findUsers, findUserById 메서드를 리팩토링한다.
- Dto의 값을 꺼내서 테스트하도록 변경한다.
- UserControllerTest의 테스트 메서드명을 변경한다.
- 메서드의 파라미터에 final 키워드를 추가한다.
- 예외 메시지를 상수로 분리한다.

workromancer reviewed (approved):
오늘 하루 고생 많으셨습니다. ㅎㅎ
